src/Agent.Plugins/BuildArtifact/BuildArtifactPluginV1.cs

```diff
@@ -0,0 +1,382 @@
+            string specificBuildWithTriggering = context.GetInput(TaskProperties.SpecificBuildWithTriggering, required: false);
+            string buildVersionToDownload = context.GetInput(TaskProperties.BuildVersionToDownload, required: false);
+            string targetPath = context.GetInput(TaskProperties.DownloadPath, required: true);
+            string environmentBuildId = context.Variables.GetValueOrDefault(BuildVariables.BuildId)?.Value ?? string.Empty; // BuildID provided by environment.
+            string itemPattern = context.GetInput(TaskProperties.ItemPattern, required: false);
+            string projectName = context.GetInput(TaskProperties.Project, required: false);
+            string tags = context.GetInput(TaskProperties.Tags, required: false);
+            string allowPartiallySucceededBuilds = context.GetInput(TaskProperties.AllowPartiallySucceededBuilds, required: false);
+            string allowFailedBuilds = context.GetInput(TaskProperties.AllowFailedBuilds, required: false);
+            string allowCanceledBuilds = context.GetInput(TaskProperties.AllowCanceledBuilds, required: false);
+            string userSpecifiedBuildId = context.GetInput(TaskProperties.BuildId, required: false);
+            string defaultWorkingDirectory = context.Variables.GetValueOrDefault("system.defaultworkingdirectory").Value;
+            string downloadType = context.GetInput(TaskProperties.DownloadType, required: true);
+            // advanced
+            string retryDownloadCount = context.GetInput(TaskProperties.RetryDownloadCount, required: false);
+            string parallelizationLimit = context.GetInput(TaskProperties.ParallelizationLimit, required: false);
+            string checkDownloadedFiles = context.GetInput(TaskProperties.CheckDownloadedFiles, required: false);
+
+            targetPath = Path.IsPathFullyQualified(targetPath) ? targetPath : Path.GetFullPath(Path.Combine(defaultWorkingDirectory, targetPath));
+
+            string[] minimatchPatterns = itemPattern.Split(
+                new[] { "\n" },
+                StringSplitOptions.RemoveEmptyEntries
+            );
+
+            string[] tagsInput = tags.Split(
+                new[] { "," },
+                StringSplitOptions.None
+            );
+
+            if (!bool.TryParse(allowPartiallySucceededBuilds, out var allowPartiallySucceededBuildsBool))
+            {
+                allowPartiallySucceededBuildsBool = false;
+            }
+            if (!bool.TryParse(allowFailedBuilds, out var allowFailedBuildsBool))
+            {
+                allowFailedBuildsBool = false;
+            }
+            if (!bool.TryParse(allowCanceledBuilds, out var allowCanceledBuildsBool))
+            {
+                allowCanceledBuildsBool = false;
+            }
+            var resultFilter = GetResultFilter(allowPartiallySucceededBuildsBool, allowFailedBuildsBool, allowCanceledBuildsBool);
```

Comment:
Since this matches PipelinesArtifact, should we consider putting it somewhere common?